x/poolmanager/taker_fee.go

```diff
@@ -144,33 +144,39 @@ func (k Keeper) chargeTakerFee(ctx sdk.Context, tokenIn sdk.Coin, tokenOutDenom
 	takerFeeAmtRemaining := takerFeeCoin.Amount
 	if takerFeeCoin.Denom == defaultTakerFeeDenom {
 		// Community Pool:
-		if poolManagerParams.TakerFeeParams.OsmoTakerFeeDistribution.CommunityPool.GT(osmomath.ZeroDec()) {
+		if poolManagerParams.TakerFeeParams.OsmoTakerFeeDistribution.CommunityPool.GT(osmomath.ZeroDec()) && takerFeeAmtRemaining.GT(osmomath.ZeroInt()) {
```

Comment:
Notice I added && takerFeeAmtRemaining.GT(osmomath.ZeroInt())

I noticed in some local tests that its possible for there to be zero taker fee amt when taker fee for denom is 0, despite the fee distribution percent being non zero. This doesn't really do anything bad, but does include a zero entry to the tracker. Again, not likely on prod and even if it did happen its a non issue, but was just not clean.